### ISSUE TITLE
Why cpplint.py doesn't check #include after #ifdef/#ifndef/#endif

### ISSUE THREAD

**Issue by wylazy** [google/styleguide#249]
I Found the code snippet in cpplint.py
  ```
  # Reset include state across preprocessor directives.  This is meant`
  # to silence warnings for conditional includes.`
  match = Match(r'^\s*#\s*(if|ifdef|ifndef|elif|else|endif)\b', line)`
  if match:`
    include_state.ResetSection(match.group(1))
```
So that cpplint.py think the code below is correct
```
#include <cstring>
#include <iostream>
#ifdef HAVE_SYS_UTSNAME_H
#include <sys/utsname.h>
#endif
#include <fcntl.h>
```
I think it should report the error 'Found C system header after C++ system header. build/include_order'

**Comment by vapier** [comment]
public cpplint development in this repository has been halted (see https://github.com/google/styleguide/pull/837). please migrate to https://github.com/cpplint/cpplint now.
